### PR TITLE
Improve Typescript Types for Svelte Components

### PR DESCRIPTION
**Pull request by JeremyMoeglich**
This allows for autocompletion and type checking on Props and Events on the Viewer and Editor Component

image

This will for example lead to an Error when passing a number to the value field or when missing a field

### CODE REVIEW

File: packages/bytemd/src/editor.svelte.ts [hunk 1]
```diff
@@ -1 +1,15 @@
...
+        change: CustomEvent<{value: string}>
+    };
+    slots: {};
+};
+export declare type OutConnectorProps = typeof __propDef.props;
+export declare type OutConnectorEvents = typeof __propDef.events;
+export declare type OutConnectorSlots = typeof __propDef.slots;
+export default class OutConnector extends SvelteComponentTyped<OutConnectorProps, OutConnectorEvents, OutConnectorSlots> {
+}
+export {};
```
Comment (pd4d10) [inline packages/bytemd/src/editor.svelte.ts]: Thanks for the contribution!

Does this piece of code handwritten or automatically generated?

Comment (JeremyMoeglich) [reply packages/bytemd/src/editor.svelte.ts]: Sveltekit package can generate this automatically, I just copied the declarations from another Project and manually changed the values

Comment (pd4d10) [reply packages/bytemd/src/editor.svelte.ts]: OK. If it is generated, perhaps we can add a script to automate it?

`scripts/postinstall.mjs` would be a good place to add it.

Comment (JeremyMoeglich) [reply packages/bytemd/src/editor.svelte.ts]: I'll try to do that

Comment (JeremyMoeglich) [reply packages/bytemd/src/editor.svelte.ts]: Added script to generate `.svelte.ts` into `postinstall.mjs`

**Comment by JeremyMoeglich** [comment]
Are there any other changes that I should make?

**Pull request merged**